src/index.ts

```diff
@@ -537,10 +539,12 @@ export class Bigtable {
     this.projectName = `projects/${this.projectId}`;
     this.shouldReplaceProjectIdToken = this.projectId === '{{projectId}}';
 
-    if (options.metricsEnabled === false) {
-      this.metricsEnabled = false;
+    if (options.metricsEnabled === true) {
+      this._metricsConfigManager = new ClientSideMetricsConfigManager([
+        new GCPMetricsHandler(options as ClientOptions),
+      ]);
     } else {
-      this.metricsEnabled = true;
+      this._metricsConfigManager = new ClientSideMetricsConfigManager([]);
```

Comment:
why do we still create the MetricsConfigManager when metrics are not enabled? I think we can bypass it entirely.

Reply:
We should loop @daniel-sanche in this discussion. My original design would ignore any metrics collection if metrics aren't enabled, but [this discussion](https://github.com/googleapis/nodejs-bigtable/pull/1605#discussion_r2091929506) makes a case for a different option.

By the way I think we should just merge this PR and continue comments on https://github.com/googleapis/nodejs-bigtable/pull/1571. This PR was just meant to separate some changes Daniel and I wanted to talk about from the rest of the readRows stuff, but now that he has approved it will be easier to discuss everything in the other PR.